### DISCUSSION TITLE
Are SSH key agents supported?

### DISCUSSION THREAD

**Discussion by b2sc** [wavetermdev/waveterm#131] [General]
My local session "sees" my private keys (ssh-add -l). Still, SSH connections can not be created without providing the private key file. But my key is on a YubiKey. ;)

**Comment by sawka** [comment]
Unfortunately ssh-agent is not supported for native Wave Connections.

Your local session "sees" the keys because that bash session is run through the normal login .bash_profile flow.  But wavesrv (which handles creating the remote SSH connections) might not pick that up (it is run via a `bash -c`).

So, two responses here.  First, I'm wondering if we could run wavesrv using a login shell instead (which might just fix your issue).  But longer term, we plan on building ssh agent functionality directly into Wave.  Then we can also support things like YubiKey, 1password, and keychain natively.

**Reply by b2sc** [reply]
Found a solution. If you start Wave via `open /Applications/Wave.app` it gets the needed environment and the GPG key agent (needed for Yubikey) is used. Probably works for other SSH key agents, too.
